core/trino-spi/src/main/java/io/trino/spi/ptf/DescriptorArgumentSpecification.java

```diff
@@ -16,13 +16,37 @@
 public class DescriptorArgumentSpecification
         extends ArgumentSpecification
 {
-    public DescriptorArgumentSpecification(String name)
+    private DescriptorArgumentSpecification(String name, boolean required, Descriptor defaultValue)
     {
-        super(name, true, null);
+        super(name, required, defaultValue);
     }
 
-    public DescriptorArgumentSpecification(String name, Descriptor defaultValue)
+    public static Builder descriptorArgumentBuilder(String name)
```

Comment:
I think putting required fields in a `builder()` constructor kind of defeats of the purposes of a builder, which is readability ("named parameters"). I know we do so in _some_ places, but i don't think we should, especially for code that's written very rarely and read much more often, as the one here.

Reply:
I saw that pattern in Trino and found it good because it keeps the usage concise. Also, it gives the notion that `name` is not optional, while the other field is.

Reply:
You can enforce optionality by throwing in `build()`. Yet I came ok with SINGLE argument passed to `builder()` method as we have here.

Reply:
It is enforced anyway by throwing in the `DescriptorArgumentSpecification` constructor. I wasn't trying here to enforce the semantics, but to convey it clearly to the user of the builder. If you still think that builder shouldn't take `name` in the constructor, I'll remove it.

Reply:
Up to me it is fine to have this arg (as stated above) :)

Reply:
@findepi is it acceptable for you to leave the argument?

Reply:
While rebasing, I decided to remove the argument `name` from the Builder constructor. The code didn't render good that way.